src/mapping/attractor_mapping_recurrences.jl

```diff
@@ -210,21 +235,22 @@ Base.@kwdef struct RegularGrid{D} <:Grid
     grid_steps::SVector{D, Float64}
     grid_minima::SVector{D, Float64}
     grid_maxima::SVector{D, Float64}
+    grid
 end
 
 struct IrregularGrid{D} <:Grid
     grid::NTuple{D,Vector{Float64}}
 end
 
-Base.@kwdef struct IrregularGridViaMatrix{D} <:Grid
+Base.@kwdef struct SubdivisionBasedGrid{D} <:Grid
     grid_steps
     grid_minima::SVector{D, Float64}
     grid_maxima::SVector{D, Float64}
-    matrix::Array
+    matrix
     grid
```

Comment:
Why do you need the grid as a field? In any case, if you need it everything should be typed. `grid` must be typed, and also `grid_steps` must be typed. I don't udnerstand why there are things that aren't typed.